Source/Core/Core/IOS/USB/Emulated/Skylander.cpp

```diff
@@ -0,0 +1,804 @@
+        // Response	{ 'A', (00 | 01),
+        // ff, 77, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00,
+        // 00, 00, 00, 00, 00, 00, 00, 00 }
+        // The 2nd byte of the command is whether to activate (0x01) or deactivate (0x00) the
+        // portal. The response echos back the activation byte as the 2nd byte of the response. The
+        // 3rd and 4th bytes of the response appear to vary from wired to wireless. On wired
+        // portals, the bytes appear to always be ff 77. On wireless portals, during activation the
+        // 3rd byte appears to count down from ff (possibly a battery power indication) and during
+        // deactivation ed and eb responses have been observed. The 4th byte appears to always be 00
+        // for wireless portals.
+
+        // Wii U Wireless: 41 01 f4 00 41 00 ed 00 41 01 f4 00 41 00 eb 00 41 01 f3 00 41 00 ed 00
+        if (cmd->length == 2)
+        {
+          data = {buf[0], buf[1]};
+          result = {0x41, buf[1], 0xFF, 0x77, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    0x00, 0x00,   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    0x00, 0x00,   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+          m_queries.push(result);
+          expected_count = 10;
+          system.GetSkylanderPortal().Activate();
+        }
+        break;
+      }
+      case 'C':
+      {
+        // Color
+        // Command	{ 'C', 12, 34, 56, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00,
+        // 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00 }
+        // Response	{ 'C', 12, 34, 56, 00, 00,
+        // 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00,
+        // 00, 00, 00, 00 }
+        // The 3 bytes {12, 34, 56} are RGB values.
+
+        // This command should set the color of the LED in the portal, however this appears
+        // deprecated in most of the recent portals. On portals that do not have LEDs, this command
+        // is silently ignored and do not require a response.
+        if (cmd->length == 4)
+        {
+          system.GetSkylanderPortal().SetLEDs(0x01, buf[1], buf[2], buf[3]);
+          data = {0x43, buf[1], buf[2], buf[3]};
+          expected_count = 12;
+        }
+        break;
+      }
+      case 'J':
+      {
+        // Sided color
+        // The 2nd byte is the side
+        // 0x00: right
+        // 0x01: left and right
+        // 0x02: left
+
+        // The 3rd, 4th and 5th bytes are red, green and blue
+
+        // The 6th byte is unknown. Observed values are 0x00, 0x0D and 0xF4
+
+        // The 7th byte is the fade duration. Exact value-time corrolation unknown. Observed values
+        // are 0x00, 0x01 and 0x07. Custom commands show that the higher this value the longer the
+        // duration.
+
+        // Empty J response is send after the fade is completed. Immeditately sending it is fine
```

Comment:
```suggestion
        // Empty J response is sent after the fade is completed. Immeditately sending it is fine
```

Reply:
also immeditately should be immediately

Reply:
I'm actually going to rename some of those variables, because we don't immediately send some of those results